File: awa/sdk_dist/csharp/Awa.Client/Utilities/Workflow/ApplySingleFileDiffWorkflow.cs
using System;
using System.Collections.Generic;
using System.Threading.Tasks;
using Awa.Client.Constants;

namespace Awa.Client.Utilities.Workflow
{
    public static partial class WorkflowUtilities
    {
        /// <summary>
        /// Apply a diff to a single file based on a natural language prompt.
        /// </summary>
        /// <param name="filePath">The path to the file to modify.</param>
        /// <param name="prompt">The natural language request describing the changes to make.</param>
        public static async Task ApplySingleFileDiffWorkflow(string filePath, string prompt)
        {
            var args = new Dictionary<string, object?>
            {
                ["file_path"] = filePath,
                ["natural_language_request"] = prompt
            };

            await Temporalio.Workflows.Workflow.ExecuteChildWorkflowAsync(
                AwaConstants.WorkflowApplySingleFileDiff,
                new object?[] { args },
                new()
                {
                    TaskQueue = AwaConstants.AwaDefaultTaskQueue
                });
        }
    }
}

File: awa/awa/sdk/utils/workflow/apply_single_file_diff_workflow.py
from temporalio import workflow

from awa.sdk import constants


async def apply_single_file_diff_workflow(file_path: str, prompt: str) -> None:
    """Apply a diff to a single file based on a natural language prompt.

    Args:
        file_path: The path to the file to modify.
        prompt: The natural language request describing the changes to make.

    """
    await workflow.execute_child_workflow(
        workflow=constants.WORKFLOW_APPLY_SINGLE_FILE_DIFF,
        arg={
            "file_path": file_path,
            "natural_language_request": prompt,
        },
        task_queue=constants.AWA_DEFAULT_TASK_QUEUE,
    )
